rotkehlchen/db/taxable_events.py

```diff
@@ -0,0 +1,348 @@
+        self.db.conn_transient.commit()
+        return identifier
+
+    def add_event(self, report_id: int, time: Timestamp, event: dict) -> None:
+        """Adds a new event to a transient report for the PnL history in a given time range
+
+        May raise:
+        - sqlcipher.IntegrityError if there is a conflict at serialization of the event
+        """
+        cursor = self.db.conn_transient.cursor()
+        query = """
+        INSERT INTO pnl_events(
+            report_id, timestamp, data
+        )
+        VALUES(?, ?, ?);"""
+        cursor.execute(query, (report_id, time, serialize_event_to_db(event)))
+        self.db.conn_transient.commit()
+
+    def add_events(self, report_id: int, events: List[Dict[str, Any]]) -> None:
+        """Adds taxable events to the database"""
+        event_tuples: List[Tuple[Any, ...]] = []
+        for event in events:
+            event_tuples.append((
+                report_id,
+                event['time'],
+                serialize_event_to_db(event),
+            ))
+
+        query = """
+            INSERT INTO pnl_events(
+              report_id,
+              timestamp,
+              event
+            )
+            VALUES (?, ?, ?)
+        """
+        self.db.write_tuples(
+            tuple_type='pnl_event',
+            query=query,
+            tuples=event_tuples,
+        )
+
+    def get_or_query_report_events(
+            self,
+            report_id: int,
+            start_ts: Timestamp,
+            end_ts: Timestamp,
+    ) -> Tuple[List[Dict[str, Any]], Optional[Timestamp], Optional[Timestamp]]:
+        """Only queries cached events and prepares them for the response"""
+        ranges = DBQueryRanges(self.db)
+        ranges_to_query = ranges.get_location_query_ranges(
+            location_string=f'pnl_events_{report_id}',
+            start_ts=start_ts,
+            end_ts=end_ts,
+        )
+        cache_data = []
+        query_start_ts = None
+        query_end_ts = None
+        for query_start_ts, query_end_ts in ranges_to_query:
+            try:
+                cache_data = self.get_events(filter_=ReportsFilterQuery.make(report_id=report_id))
+            except sqlcipher.DatabaseError as e:  # pylint: disable=no-member
+                self.msg_aggregator.add_error(
+                    f'Got error "{str(e)}" while querying the reports cache '
+                    f'from rotki. Events not added to the DB '
+                    f'from_ts: {query_start_ts} '
+                    f'to_ts: {query_end_ts} ',
+                )
+
+        # and also set the last queried timestamps for the report
+        ranges.update_used_query_range(
+            location_string=f'pnl_events_{report_id}',
+            start_ts=start_ts,
+            end_ts=end_ts,
+            ranges_to_query=ranges_to_query,
+        )
+        log.debug(f"___cachedata___ {cache_data[0]}")
+
+        return cache_data, query_start_ts, query_end_ts
```

Comment:
`query_start_ts`, `query_end_ts` here is going to be either `None` and `None` or the last iteration of the for loop above. So the question here is .. why? Why do you do that and what possible purpose can this serve?

I just don't get this function. The ranges_to_query functionality is ONLY for ranges that have not been queried yet, and as such not saved in the DB. It's to query external data. Your PR should only deal with internal data and caching them. This whole thing is not needed.